server/neptune/workflows/internal/activities/github.go

```diff
@@ -204,3 +204,40 @@ func (a *githubActivities) FetchRoot(ctx context.Context, request FetchRootReque
 		LocalRoot: localRoot,
 	}, nil
 }
+
+type CompareCommitRequest struct {
+	DeployRequestRevision  string
+	LatestDeployedRevision string
+	Repo                   internal.Repo
+}
+
+type CompareCommitResponse struct {
+	DeployRequestRevisionAheadBy int
+}
+
+func (a *githubActivities) CompareCommit(ctx context.Context, request CompareCommitRequest) (CompareCommitResponse, error) {
+	client, err := a.ClientCreator.NewInstallationClient(request.Repo.Credentials.InstallationToken)
+	if err != nil {
+		return CompareCommitResponse{}, errors.Wrap(err, "creating installation client")
+	}
+
+	comparison, resp, err := client.Repositories.CompareCommits(ctx, request.Repo.Owner, request.Repo.Name, request.LatestDeployedRevision, request.DeployRequestRevision, &github.ListOptions{})
+	if err != nil {
+		return CompareCommitResponse{}, errors.Wrap(err, "comparing commits")
+	}
+
+	if comparison.GetStatus() == "" || resp.StatusCode != http.StatusOK {
+		return CompareCommitResponse{}, errors.New("invalid commit comparison status")
+	}
+
+	// aheadBy is 0 when behindBy gt 0. We subtract to give a negative value which indicates the deploy request revision is behind
+	aheadBy := comparison.GetAheadBy()
+	behindBy := comparison.GetBehindBy()
+	if behindBy > 0 {
+		aheadBy -= behindBy
+	}
+
```

Comment:
Why are you doing it this way? Can't you just use the status?

ie. 
https://github.com/lyft/deployapi/blob/76de5a28d996c3805ab4753def02f4aae7a82051/pkg/github/github.go#L267